### PR TITLE
DM-36772: Migrate PostgreSQL credentials to mounted file

### PR DESCRIPTION
**Pull request by kfindeisen**
This PR removes the `make_pgpass` module entirely, and uses the extra freedom to switch the APDB access code in `MiddlewareInterface` to a single URL. It updates the Playbook to reflect the reduced number of input variables as well as our migration from Slaclab to Argo CD.

### CODE REVIEW

File: doc/playbook.rst [hunk 1]
```diff
@@ -214,37 +214,26 @@ The service can be controlled with ``kubectl`` from ``rubin-devl``.
...
-* RUBIN_INSTRUMENT: the "short" instrument name
-* PIPELINES_CONFIG: a machine-readable string describing which pipeline(s) should be run for which visits.
-  Notation is complex and still in flux; see :file:`../python/activator/config.py` for current documentation and examples.
-* PUBSUB_VERIFICATION_TOKEN: choose an arbitrary string matching the Pub/Sub endpoint URL below.
-  This variable is currently unused and may be removed in the future.
-* IMAGE_BUCKET: bucket containing raw images
-* CALIB_REPO: URI to repo containing calibrations (and templates)
-* LSST_DISABLE_BUCKET_VALIDATION: set this so to disable validation of S3 bucket names, allowing Ceph multi-tenant colon-separated names to be used.
-* URL_APDB: the URL to the APDB, in any form recognized by SQLAlchemy
-* KAFKA_CLUSTER: hostname and port of the Kafka provider
```
Comment (hsinfang) [inline doc/playbook.rst:242]: Thank you for moving these docs to phalanx!

**Review by hsinfang — APPROVED** [review]

File: doc/playbook.rst [hunk 2]
```diff
@@ -234,19 +234,14 @@ It includes the following required environment variables:
 * CALIB_REPO: URI to repo containing calibrations (and templates)
 * LSST_DISABLE_BUCKET_VALIDATION: set this so to disable validation of S3 bucket names, allowing Ceph multi-tenant colon-separated names to be used.
 * IP_APDB: IP address or hostname and port of the APDB (see `Databases`_, below)
-* IP_REGISTRY: IP address or hostname and port of the registry database (see `Databases`_)
```
[the remark this answers is not in the document]
Comment (kfindeisen) [reply doc/playbook.rst:237]: Well, the catch is that this version of the code won't work on Google (we can't get a `.pgpass` file with the right permissions there). So if you're running on Google, you must be using a version that still uses those variables. 😵‍💫

**Comment by kfindeisen** [comment]
I pushed a revised version of the Argo CD instructions -- it turns out you *can* redeploy purely from Argo, as long as you don't have any `phalanx` changes to sync. 🙂

File: doc/playbook.rst [hunk 3]
```diff
@@ -214,44 +214,29 @@ The service can be controlled with ``kubectl`` from ``rubin-devl``.
...
+  Scroll down to the live manifest, click "edit", then update the ``template.spec.containers.image`` key to point to the new service container (likely a ticket branch instead of ``latest``).
+  The service will immediately redeploy with the new image.
+  To force an update of the container, edit ``template.metadata.annotations.revision``.
+  *Do not* click "SYNC" on the main screen, as that will undo all your edits.
+* If you will be making permanent changes of any kind, the above procedure would force you to re-enter your changes with each update of the ``phalanx`` branch.
+  Instead, clone the `lsst-sqre/phalanx`_ repo and navigate to the ``applications/prompt-proto-service-<instrument>`` directory.
+  Edit ``values-usdfdev-prompt-processing.yaml`` to point to the new service container (likely a ticket branch instead of ``latest``) and push the branch.
+  You do not need to create a PR.
+  Then, in the Argo UI, select the service, click "SYNC" from the toolbar, and enter the branch name under "Revision".
```
Comment (hsinfang) [inline doc/playbook.rst:231]: Might reference https://phalanx.lsst.io/developers/deploy-from-a-branch.html#switching-the-argo-cd-application-to-sync-the-branch for this

Comment (kfindeisen) [reply doc/playbook.rst:231]: Thanks for pointing that out! I didn't know about the "Target Revision" field. I wonder how permanent it is, compared to "Sync" → "Revision"?

Comment (hsinfang) [reply doc/playbook.rst]: Looks to me that the  "Sync" → "Revision" is a one time syncing to the specified branch.  If that's different from the "Target Revision" field in the App Detail Summary, it'll show as "Out of sync" in its sync status.  If the "Target Revision" field is the ticket branch the sync status follows the ticket branch. 

Either sounds fine to me. In the end of the development work the target revision will be `main` either way.

**Pull request merged**